src/gretel_trainer/trainer.py

```diff
@@ -0,0 +1,156 @@
+"""Main Trainer Module"""
+
+from collections import namedtuple
+from enum import Enum
+import json
+import logging
+import pandas as pd
+import pkgutil
+import yaml
+
+from gretel_client import configure_session, ClientConfig
+from gretel_client.projects import create_or_get_unique_project
+from gretel_client.projects.models import read_model_config
+from gretel_client.projects.jobs import Status
+from gretel_synthetics.utils.header_clusters import cluster
+
+from gretel_trainer import strategy, runner
+
+
+logger = logging.getLogger(__name__)
+logger.setLevel("DEBUG")
+
+
+class ExtendedEnum(Enum):
+    """Utility class for Model enum"""
+
+    @classmethod
+    def get_types(cls):
+        return list(map(lambda c: c.name, cls))
+
+    @classmethod
+    def get_config(cls, model: str):
+        return cls[model].config
+
+
+class Model(namedtuple("Model", "config"), ExtendedEnum):
+    """Enum to pair valid models and configurations"""
+
+    GretelLSTM = "templates/gretel_lstm.yaml"
```

Comment:
Qq: any reason not to use the default configs we have on the blueprints repo and just access them by their namespace like you can now with the client? This feels like we are replicating default configs across different repos. Are these configs more optimized for trainer usage?

Reply:
Modified the trainer to use configs loaded from the blueprints repo